site/frontend/src/pages/compare/artifact-size/artifact-size-table.vue

```diff
@@ -65,6 +65,29 @@ function getClass(a: number | undefined, b: number | undefined): string {
   }
   return diffClass(b - a);
 }
+
+function generateTitle(component: string): string {
+  if (component === "rustc") {
+    return `Main binary of the Rust compiler. It is a small shim that links to librustc_driver.so,
+ which contains most of the compiler logic.`;
+  } else if (component === "rustdoc") {
+    return `Tool that generates documentation of Rust crates. Links to librustc_driver.so, which provides
+ most of the compiler logic.`;
+  } else if (component === "cargo") {
+    return "Build tool that compiles crates and their dependencies.";
```

Comment:
Do we want the tooltips to be what the tool is, or what the artifact is ? I'm hoping people won't learn what LLVM or cargo is from this tooltip, so e.g. "Build tool that compiles crates and their dependencies." vs "cargo's executable. Will be the one from the default toolchain if not present in a locally built toolchain" (just an example).

I also wonder if we should add the little "?" icon to show there's a tooltip, like we do elsewhere.

![image](https://github.com/rust-lang/rustc-perf/assets/247183/23876cab-c642-4021-b6c3-eae2078931cc)

Reply:
Again, very good points :)